[[ Bug 20366 ]] Add edition icon and upgrade button to docs

montegoulding opened the pull request:
This patch uses the edition element to add an icon to each non-community
API in the search results list. The same icon use used in the edition
element. If the API is not available in the current edition an upgrade
button is shown.

livecodeali commented on Documentation/html_viewer/js/dictionary_functions.js:
```diff
@@ -483,7 +486,29 @@
+		
+		if (tUpgrade)
+		{
+			tHTML += ' <a href="#" class="btn btn-default" onclick="javascript:liveCode.showUpgradeOptions(\'' 
```
Could you make `upgrade` a class and `edition` and `display_name` attributes of the element, and then add an 
```
 		$("body").on("click", ".upgrade", function() {
 			var tEdition = $(this).attr("edition");
                        ...
 		});
```
to the `setActions` function please? It makes it much easier to see what is happening, modify and add other places that need to show upgrade options if we want them

livecodeali replied:
Plus you can gate the call to `showUpgradeOptions` with `isRunningInLiveCodeBrowser()` so there's no error in an external browser

montegoulding replied:
@livecodeali sorry have had my head down for a few days and just read this. Yes to the first comment. Yes to the second but it makes me wonder if we could build the URL locally so the button can work when in an external browser.

montegoulding replied:
OK so I'm just redirecting them to livecode.com for now so it doesn't do nothing if not running in the browser widget.

livecodeali replied:
That's a good idea about making the button work in an external browser yeah...  could just redirect here: https://livecode.com/products/livecode-platform/pricing/

montegoulding committed:
[[ Bug 20366 ]] Add edition icon and upgrade button to docs

This patch uses the edition element to add an icon to each non-community
API in the search results list. The same icon use used in the edition
element. If the API is not available in the current edition an upgrade
button is shown.

livecodeali commented:
@livecode-vulcan review ok 17434f2

livecode-vulcan commented:
💙 review by @livecodeali ok 17434f293789f2c89757b2907c65685ab06ad3d1
